sourcegraph/sourcegraph
# JetBrains: move all network traffic to agent

Previously, the JetBrains plugin still sent network requests from the Java side for telemetry and to test whether the provided connection parameters were valid. This PR replaces this custom network traffic with requests to the agent.

## Test plan

* Update Cody to PR https://github.com/sourcegraph/cody/pull/734
* `./gradlew -PforceAgentBuild=true :runIDE`
* `tail -f build/sourcegraph/cody-agent-trace.json`
* Trigger autocomplete, confirm it logs an event like this

```
[Trace - 02:49:58 PM] Sending request 'graphql/logEvent - (38)'
Params: {
  "event": "CodyJetBrainsPlugin:completion:suggested",
  "userCookieID": "7d9607a1-d84c-4329-8a11-c1392676ec33",
  "url": "",
  "source": "IDEEXTENSION",
  "referrer": "JETBRAINS",
  "argument": {
    "latency": 30,
    "displayDuration": 3003,
    "isAnyKnownPluginEnabled": false,
    "serverEndpoint": "https://sourcegraph.com/"
  },
  "publicArgument": {
    "latency": 30,
    "displayDuration": 3003,
    "isAnyKnownPluginEnabled": false,
    "serverEndpoint": "https://sourcegraph.com/"
  },
  "deviceID": "7d9607a1-d84c-4329-8a11-c1392676ec33"
}
```
* Open settings, change from s2/dotcom, press "apply", confirm it sends requests like this

```
[Trace - 02:50:22 PM] Sending request 'graphql/currentUserId - (44)'
Params: null
[Trace - 02:50:23 PM] Received response 'graphql/currentUserId - (43)' in 1303ms
Result: "REDACTED"
Error: null
```
